pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/kmp/factories.ios.kt

```diff
@@ -26,7 +58,203 @@ actual fun createEventListener(
+private fun createMessageActionResult(from: PubNubMessageActionObjC?): PNMessageActionResult {
+    return PNMessageActionResult(
+        result = BasePubSubResult(
+            channel = from!!.channel(),
+            subscription = from.subscription(),
+            timetoken = from.actionTimetoken().toLong(),
+            userMetadata = null,
+            publisher = from.publisher()
+        ),
+        event = from.event(),
+        data = PNMessageAction(
+            type = from.actionType(),
+            value = from.actionValue(),
+            messageTimetoken = from.messageTimetoken().toLong()
+        )
+    )
+}
+
+@OptIn(ExperimentalForeignApi::class)
+private fun createPresenceEventResults(from: List<*>?): List<PNPresenceEventResult> {
+    return (from as? List<PubNubPresenceEventResultObjC>)?.let {
+        it.map { item: PubNubPresenceEventResultObjC ->
+            PNPresenceEventResult(
+                event = item.event(),
+                uuid = item.uuid(),
+                timestamp = item.timetoken()?.longValue,
+                occupancy = item.occupancy()?.intValue,
+                state = item.state() as JsonElement,
+                channel = item.channel() ?: "",
+                subscription = item.subscription(),
+                timetoken = item.timetoken()?.longValue,
+                join = item.join() as? List<String>,
+                leave = item.leave() as? List<String>,
+                timeout = item.timeout() as? List<String>,
+                hereNowRefresh = item.refreshHereNow()?.boolValue,
+                userMetadata = item.userMetadata()
+            )
+        }
+    } ?: emptyList()
+}
+
+@OptIn(ExperimentalForeignApi::class)
+private fun createFileEventResult(from: PubNubFileEventResultObjC?): PNFileEventResult {
+    return PNFileEventResult(
+        channel = from!!.channel(),
+        timetoken = from.timetoken()?.longValue,
+        publisher = from.publisher(),
+        message = from.message(),
+        jsonMessage = from.message() as JsonElement,
+        file = PNDownloadableFile(
+            id = from.file().id(),
+            name = from.file().name(),
+            url = from.file().url()
+        )
+    )
+}
+
+@OptIn(ExperimentalForeignApi::class)
+private fun createObjectEvent(from: PubNubObjectEventResultObjC?): PNObjectEventResult? {
+    return mapObjectMessage(from?.message())?.let {
+        PNObjectEventResult(
+            result = BasePubSubResult(
+                channel = from!!.channel(),
+                subscription =  from.subscription(),
+                timetoken = from.timetoken()?.longValue,
+                userMetadata = from.userMetadata() as? JsonElement,
+                publisher = from.publisher(),
+            ), extractedMessage = it
+        )
+    }
+}
+
+@OptIn(ExperimentalForeignApi::class)
+private fun mapObjectMessage(from: PubNubObjectEventMessageObjC?): PNObjectEventMessage? {
+    if (from is PubNubSetUUIDMetadataEventMessageObjC) {
```

Comment:
this can be a `when` instead of `if` (IntelliJ can probably make the change automatically through context actions), but we can also fix that later

Reply:
Sure, I'll submit this change as a part of next PR